tmt/steps/provision/__init__.py

```diff
@@ -1268,6 +1276,25 @@ def is_ready(self) -> bool:
         # Enough for now, ssh connection can be created later
         return self.guest is not None
 
+    def setup(self) -> None:
+        if self.is_dry_run:
+            return
+        if self.guest is None and not self.is_dry_run:
+            raise tmt.utils.GeneralError('The guest is not available.')
+        if not self.facts.is_superuser:
+            parent = cast(Provision, self.parent)
+            ssh_command: tmt.utils.Command = self._ssh_command()
+
+            remote_commands: ShellScript = ShellScript(f'sudo dnf install -y acl')
```

Comment:
Instead of `dnf`, `guest.facts.package_manager` should be used.

(We really, really should have "install me this" helpers, because tmt installs more and more tools for its own needs, and every single place needs to take care of picking the right manager, right `sudo`, right options... Meh).

Reply:
Another point: are annotations necessary? Both mypy and pyright should be able to infer the type from the assignment alone.

Reply:
Yes, I will be using that utility. This is definitely a cleanup to do. I'll remove the type there as well.

Reply:
@happz What seems better - Allow/use `PrepareInstall` to be run in almost any step or dig the 'dnf install' part from it into `tmt.utils` and use it from there?

For unrelated dist-git thingy I'll need to install few things as well :D

Reply:
To me, it seemed better to generalize and convert `PrepareInstall` to use the generalized code.

Somewhere in my stash lies a draft of `tmt.package_managers` where I tried to cut things out. I wouldn't mind finishing it - e.g. the `avc` check needs `sestatus`, and I wasn't sure it's safe to rely on it being omnipresent. Plus we already install `rsync`, for example, so there is a need for "install me a package" primitive.

Then there is @thrix & his all-mighty install plugin effort, for which I was hoping to have this general library ready...

Now when the parallelization works are almost done, I guess it wouldn't be a bad thing to refresh this patch, there seems to be interest in it from multiple sides. And I still want to run full tmt test suite on my Ubuntu VM, soooo... `apt-get`, here we come :)